### PR TITLE
Modernize color picker and add `ColorFormField`

### PR DESCRIPTION
**Pull request by Gravatronics**
Best reviewed commit by commit.

### CODE REVIEW

File: ts/WoltLabSuite/Core/ColorUtil.ts [hunk 1]
```diff
@@ -158,40 +158,162 @@ export function hexToRgb(hex: string): RGB | typeof Number.NaN {
...
+
+/**
+ * @since 5.5
+ */
+function getColorChecker(): HTMLSpanElement {
+  let colorChecker = document.getElementById("jsColorUtilColorChecker");
+  if (colorChecker === null) {
+    colorChecker = document.createElement("span");
+    colorChecker.id = "jsColorUtilColorChecker";
+    document.body.appendChild(colorChecker);
```
Comment (TimWolla) [inline ts/WoltLabSuite/Core/ColorUtil.ts:239]: Do you really need to inject the element into the DOM?

File: ts/WoltLabSuite/Core/Ui/Color/Picker.ts [hunk 2]
```diff
@@ -67,14 +42,217 @@ class UiColorPicker {
...
+          this.colorTextInput.addEventListener("keypress", (ev) => this.updateColorFromHex(ev));
+
+          content.querySelector(".formSubmit > .buttonPrimary")!.addEventListener("click", () => this.submitDialog());
+
+          if (ColorUtil.isValidColor(this.input.value)) {
+            this.setInitialColor(this.input.value);
+          } else if (this.element.dataset.color && ColorUtil.isValidColor(this.element.dataset.color)) {
+            this.setInitialColor(this.element.dataset.color);
+          } else {
+            this.setInitialColor("#FFF0");
```
Comment (TimWolla) [inline ts/WoltLabSuite/Core/Ui/Color/Picker.ts:126]: Should alpha be 0 or F?

File: ts/WoltLabSuite/Core/ColorUtil.ts (hunk 1 above)
Comment (Gravatronics) [reply ts/WoltLabSuite/Core/ColorUtil.ts:239]: Yes, for `window.getComputedStyle()`.

File: ts/WoltLabSuite/Core/Ui/Color/Picker.ts (hunk 2 above)
Comment (Gravatronics) [reply ts/WoltLabSuite/Core/Ui/Color/Picker.ts:126]: No, `0` is correct because by default no color is set.

**Review by TimWolla — APPROVED** [review]

**Pull request merged**